src/runtime/Instance.h

```diff
@@ -117,16 +118,26 @@ class Instance : public Object {
     Memory* resolveExportMemory(std::string& name);
     Global* resolveExportGlobal(std::string& name);
 
-    const VectorWithFixedSize<Function*>& functions() { return m_functions; }
+    const Function* const* functions() { return m_functions; }
 
 private:
+    Instance(Module* module);
+    ~Instance() {}
+
+    static size_t alignedSize()
+    {
+        return (sizeof(Instance) + sizeof(void*) - 1) & ~(sizeof(void*) - 1);
```

Comment:
I have a question about alignment here.
Do we have to keep the alignment explicitly?
Is there any possible cases(target platforms) in which the aligment is broken?

Other looks good to me.

Reply:
Since Instance has pointers, it is impossible for it to be not pointer aligned.

I did it this way, since it is a compile time constant, and it is in the safe side. But I can remove the align part, since you are right, it will be always pointer aligned.

Reply:
I see. I just wonder that if there is any exceptional case about this.
As you mentioned, this is good for safety, so seems pretty nice.